src/util-map.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable import/prefer-default-export */
+/**
+ * Map helpers in $util.map
+ *
+ * $util.map contains methods to help with common
+ * Map operations, such as removing or retaining
+ * items from a Map for filtering use cases.
+ */
+
+/**
+ * Makes a shallow copy of the first map, retaining only the
+ * keys specified in the list, if they are present. All other
+ * keys will be removed from the copy.
+ * @param map
+ * @param list
+ */
+export function copyAndRetainAllKeys(map: object, list: Array<string>): object {
+  return Object.entries(map).reduce((sum, [key, value]) => {
+    if (list.indexOf(key) === -1) return sum;
+    return {
+      ...sum,
+      [key]: value,
+    };
+  }, {});
```

Comment:
This works, although in this case I'd probably rename `sum` as `accumulator` given sum implies an arithmetic sum. 

However you could also use a similar approach to `copyAndRemoveAllKeys` whereby you simply loop through each value in the list and add it to a `result` object. This would be much more readable.

Reply:
Yeah, I like that suggestion.  Much more readable.  I'll make the change.